[KB] Update reusable workflows KB to accurately describe secrets behavior (#1938)
semgrep/semgrep-docs: docs/kb/semgrep-ci/github-reusable-workflows-semgrep.md

  # Schedule the CI job (this method uses cron syntax):
  schedule:
    # Please change the cron schedule to a random time to avoid load spikes on GHA.
    - cron: '24 13 * * *' # Sets Semgrep to scan every day at 13:24 UTC.
jobs:
  call-semgrep:
    uses: {ORG}/{REPO}/.github/workflows/semgrep.yml@main
    secrets: inherit
```

When using this sample configuration, be sure to update the schedule under `on` to a random time, and set repository details and path for the reusable workflow under `jobs` to match where you stored your reusable workflow.

> The `secrets: inherit` line passes the secrets from the calling workflow to the called workflow, so each calling repository must also have a `SEMGREP_APP_TOKEN` secret added. GitHub does not currently support passing secrets from a central reusable workflow (the called workflow) to the calling workflows.

Author: I wondered about that. I don't always want to link to external resources since they can change, but I think this one is probably pretty safe. And if someone did see this and click through and it wasn't true anymore, they'd probably tell us, which could be helpful. :)

## Limitations

> As described in [Set up a reusable workflow](#set-up-a-reusable-workflow), you still need to both create a `.github/workflows/semgrep.yml` file for each repository to call the reusable workflow and add a `SEMGREP_APP_TOKEN` secret to the repository. This is in contrast to [repository rulesets](/docs/kb/semgrep-ci/github-repository-rulesets-semgrep), which only require the central workflow file to be added.

Reviewer: ```suggestion
As described in [Set up a reusable workflow](#set-up-a-reusable-workflow), you must create a `.github/workflows/semgrep.yml` file for each repository to call the reusable workflow **and** add a `SEMGREP_APP_TOKEN` secret to the repository. This is in contrast to [repository rulesets](/docs/kb/semgrep-ci/github-repository-rulesets-semgrep), which only require the central workflow file to be added.
```
